File: src/main/java/com/refinedmods/refinedstorage/api/network/grid/IGridTab.java
package com.refinedmods.refinedstorage.api.network.grid;

import com.refinedmods.refinedstorage.api.render.IElementDrawer;
import com.refinedmods.refinedstorage.api.util.IFilter;
import net.minecraft.client.gui.Font;
import net.minecraft.client.gui.GuiGraphics;
import net.minecraft.world.item.ItemStack;
import net.neoforged.neoforge.fluids.FluidStack;
import java.util.List;

/**
 * Represents a grid tab.
 */
public interface IGridTab {
    int TAB_WIDTH = 28;
    int TAB_HEIGHT = 31;

    /**
     * @return the filters
     */
    List<IFilter> getFilters();

    /**
     * Draws the tooltip of this tab at the given position.
     *
     * @param font     the font
     * @param graphics the graphics
     * @param x        the x position
     * @param y        the y position
     */
    void drawTooltip(Font font, GuiGraphics graphics, int x, int y);

    /**
     * Draws the icon.
     *
     * @param graphics the graphics
     * @param x        the x position
     * @param y        the y position
     */
    void drawIcon(GuiGraphics graphics, int x, int y, IElementDrawer<ItemStack> itemDrawer, IElementDrawer<FluidStack> fluidDrawer);
}

File: src/main/kotlin/com/refinedmods/refinedstorage/api/network/grid/IGridTab.kt
package com.refinedmods.refinedstorage.api.network.grid

import com.refinedmods.refinedstorage.api.render.IElementDrawer
import com.refinedmods.refinedstorage.api.util.IFilter
import net.minecraft.client.util.math.MatrixStack
import net.minecraft.item.ItemStack
import reborncore.common.fluid.container.FluidInstance
import java.awt.font.FontRenderContext


/**
 * Represents a grid tab.
 */
interface IGridTab {
    /**
     * @return the filters
     */
    val filters: List<IFilter<*>>

    /**
     * Draws the tooltip of this tab at the given position.
     *
     * @param matrixStack  the matrix stack
     * @param x            the x position
     * @param y            the y position
     * @param screenWidth  the screen width
     * @param screenHeight the screen height
     * @param fontRenderer the font renderer
     */
    fun drawTooltip(matrixStack: MatrixStack, x: Int, y: Int, screenWidth: Int, screenHeight: Int, fontRenderer:  FontRenderContext)

    /**
     * Draws the icon.
     *
     * @param matrixStack the matrix stack
     * @param x           the x position
     * @param y           the y position
     */
    fun drawIcon(matrixStack: MatrixStack, x: Int, y: Int, itemDrawer: IElementDrawer<ItemStack>, fluidDrawer: IElementDrawer<FluidInstance>)

    companion object {
        const val TAB_WIDTH = 28
        const val TAB_HEIGHT = 31
    }
}